src/main/java/com/google/cloud/tools/gradle/appengine/appyaml/AppEngineAppYamlPlugin.java

```diff
@@ -89,10 +91,10 @@ private void configureExtensions() {
           // we can only set the default location of "archive" after project evaluation (callback)
           if (stageExtension.getArtifact() == null) {
             if (project.getPlugins().hasPlugin(WarPlugin.class)) {
-              War war = (War) project.getProperties().get("war");
+              War war = tasks.withType(War.class).getByName("war");
```

Comment:
For some reason that I don't know, the previous code was returning null.

Also I think the new code may still unnecessarily create the war task.

Reply:
Right you are. It does instantiate create task.
The way to avoid task creation is `stageExtension.setArtifact(tasks.named<War>(WarPlugin.WAR_TASK_NAME).map { it.getArchiveFile() })`

Reply:
Removed `getByName()` in another place in bae6dc2. I assume `getProject().zipTree(object)` will work fine when given a `TaskProvider`. (Previously, it was getting `File` directly from `getArchivePath()`.) Does this look good?